core/merge.js

```diff
@@ -241,14 +245,16 @@ class Merge {
   async updateFileAsync(side /*: SideName */, doc /*: Metadata */) {
     log.debug({ path: doc.path }, 'updateFileAsync')
     const { path } = doc
-    const file /*: ?Metadata */ = await this.pouch.byIdMaybe(doc._id)
+    const file /*: ?Metadata */ = await this.pouch.bySyncedPath(doc.path)
     metadata.markSide(side, doc, file)
     if (file && file.docType === 'folder') {
       throw new Error("Can't resolve this conflict!")
     }
     metadata.assignMaxDate(doc, file)
 
     if (file) {
+      // FIXME: $FlowFixMe use different types to manage records?
```

Comment:
It looks like a good idea

Reply:
And I've actually created it (https://github.com/cozy-labs/cozy-desktop/pull/1959/files/375579cf3ce69c02801f7490e5dee373a6fdc96c#diff-35b11e134833589d8cb2ae9238f6245ed0b49db0e2bbd8cc1221e4247de23ba1R32). I need to use it across the app though. I'll make a pass on the code base.